### PR TITLE
Add basic documentation

### PR DESCRIPTION
**Pull request by gfontenot**
I _think_ this format is correct, but I'm not 100% sure. Eyes on wording/format
appreciated.

ping @christoomey @gabebw @keith

### CODE REVIEW

File: doc/xcodebuild.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+xcodebuild.txt* Plugin to build and test Xcode projects from vim
```
Comment (keith) [inline doc/xcodebuild.txt]: I think there's suppose to be a leading `*` here. I feel like I went through this for [investigate](https://github.com/keith/investigate.vim/blob/master/doc/investigate.txt) ages about, but I'm pretty rusty at this point

File: doc/xcodebuild.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+==============================================================================
+CONTENTS                               *xcodebuild-contents*
+
+
+==============================================================================
+INTRODUCTION                           *xcodebuild*
+
+Xcodebuild.vim is a thin wrapper around Apple's xcodebuild command line
+application that lets you build and test your Xcode projects from within vim.
+It dynamically figures out your project information, and passes the proper
```
Comment (gabebw) [inline doc/xcodebuild.txt]: I'd kill the comma.

File: doc/xcodebuild.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+flags to xcodebuild.
+
+If you have xcpretty installed, it will use that to reformat the output from
+xcodebuild. By default, xcodebuild.vim passes the `--color` flag to xcpretty.
+
+==============================================================================
+CONFIGURATION                             *xcodebuild-configuration*
+
+* Set the flags passed xcpretty for all actions
```
Comment (gabebw) [inline doc/xcodebuild.txt]: passed TO xcpretty

Comment (gabebw) [inline doc/xcodebuild.txt]: Also: "for all actions, including test actions"?

**Comment by gabebw** [comment]
I have no idea what documentation format looks like but I do have opinions on the prose!

**Comment by gfontenot** [comment]
Thanks fellas!

**Pull request merged**